modules/registry-refactoring/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/ExperimentSummaryRepository.java

```diff
@@ -0,0 +1,265 @@
+    }
+
+    public List<ExperimentSummaryModel> searchAllAccessibleExperiments(List<String> accessibleExperimentIds, Map<String, String> filters, int limit,
+                                                                       int offset, Object orderByIdentifier, ResultOrderType resultOrderType) throws RegistryException {
+        String query = "SELECT ES FROM " + ExperimentSummaryEntity.class.getSimpleName() + " ES WHERE ";
+        Map<String, Object> queryParameters = new HashMap<>();
+
+        if (filters != null && !filters.isEmpty()) {
+
+            for (String field : filters.keySet()) {
+
+                if (field.equals(DBConstants.Experiment.USER_NAME)) {
+                    logger.debug("Filter Experiments by User");
+                    queryParameters.put(DBConstants.Experiment.USER_NAME, filters.get(field));
+                    query += "ES.userName LIKE :" + DBConstants.Experiment.USER_NAME + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.GATEWAY_ID)) {
+                    logger.debug("Filter Experiments by Gateway ID");
+                    queryParameters.put(DBConstants.Experiment.GATEWAY_ID, filters.get(field));
+                    query += "ES.gatewayId LIKE :" + DBConstants.Experiment.GATEWAY_ID + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.PROJECT_ID)) {
+                    logger.debug("Filter Experiments by Project ID");
+                    queryParameters.put(DBConstants.Experiment.PROJECT_ID, filters.get(field));
+                    query += "ES.projectId LIKE :" + DBConstants.Experiment.PROJECT_ID + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.EXPERIMENT_NAME)) {
+                    logger.debug("Filter Experiments by Name");
+                    queryParameters.put(DBConstants.Experiment.EXPERIMENT_NAME, filters.get(field));
+                    query += "ES.name LIKE :" + DBConstants.Experiment.EXPERIMENT_NAME + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.DESCRIPTION)) {
+                    logger.debug("Filter Experiments by Description");
+                    queryParameters.put(DBConstants.Experiment.DESCRIPTION, filters.get(field));
+                    query += "ES.description LIKE :" + DBConstants.Experiment.DESCRIPTION + " AND ";
+                }
+
+                else if (field.equals(DBConstants.Experiment.EXECUTION_ID)) {
+                    logger.debug("Filter Experiments by Execution ID");
+                    queryParameters.put(DBConstants.Experiment.EXECUTION_ID, filters.get(field));
+                    query += "ES.executionId LIKE :" + DBConstants.Experiment.EXECUTION_ID + " AND ";
+                }
+
+            }
+
+        }
+
+        if (filters.get(DBConstants.ExperimentStatus.STATE) != null) {
```

Comment:
How about just making the if blocks above like this one, instead of inside that for loop. I'm not sure why they are in a for loop above.

Reply:
Removed the for loop.